megamek/src/megamek/common/weapons/ArtilleryBayWeaponIndirectHomingHandler.java

```diff
@@ -474,6 +475,8 @@ protected int calcCounterAV () {
         Entity entityTarget = (Entity) target;
         // any AMS bay attacks by the target?
         ArrayList<Mounted> lCounters = waa.getCounterEquipment();
+        //We need to know how much heat has been assigned to offensive weapons fire by the defender this round
+        int weaponHeat = getLargeCraftHeat(entityTarget) + entityTarget.heatBuildup;
```

Comment:
It might just be me, but it looks like the contents of this calcCounterAV() method across all 8 source files in this code review are just about identical. Is there a way to consolidate this logic? And maybe not implement getLargeCraftHeat() 8 different times?

Reply:
I didn't find any when I wrote calcCounterAV() originally - I think there are 4 versions that are different in subtle ways, either for reporting or null checks, and since bay weapons and normal weapons follow a different tree, there have to be 8 copies.  I tried to consolidate getLargeCraftHeat at a higher level, but found it was returning 0 when it shouldn't when I did that

Reply:
I just did a diff on ArtilleryBayWeaponIndirectHomingHandler and ArtilleryWeaponIndirectHomingHandler's calcCounterAV and getLargeCraftHeat methods and only found one diff, which is the entity null check at the top of calcCounterAV. Diffs with other methods do show some more substantial differences, so consolidating the whole method may not be feasible. However, you're still duplicating a lot of logic (how many times do you invoke the code that checks if a weapon has 'heat as dice'?)

In this situation, I would suggest creating several smaller-scope helper methods to consolidate the functionality that is identical. We want to avoid getting into a situation where someone needs to make changes to this code later on and winds up either a) having to change 8 different files or b) missing some of these 8 different files and creating bugs.